### PR TITLE
Synonyms - Fixes error when no ID is specified

### PR DESCRIPTION
**Pull request by carlosdelest**
When synonyms have no ID specified, they have no ID when retrieved via GET operations. This fixes the issue.

### CODE REVIEW

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml [hunk 1]
```diff
@@ -3,6 +3,25 @@ setup:
...
+            - synonyms: "hello, hi"
+            - synonyms: "bye => goodbye"
+
+  - match: { result: "created" }
+
+  - do:
+      synonyms.get:
+        synonyms_set: test-update-synonyms
+
+  - length: { synonyms_set.0.id: 20 }
```
Comment (carlosdelest) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml:22]: Checking for length is the best I could think of to check the ID is included 🤷

File: server/src/main/java/org/elasticsearch/synonyms/SynonymRule.java [hunk 2]
```diff
@@ -43,7 +44,7 @@ public class SynonymRule implements Writeable, ToXContentObject {
     private final String id;
 
     public SynonymRule(@Nullable String id, String synonyms) {
-        this.id = id;
+        this.id = Objects.requireNonNullElse(id, UUIDs.base64UUID());
```
Comment (carlosdelest) [inline server/src/main/java/org/elasticsearch/synonyms/SynonymRule.java:47]: To make sure an ID is used, I've moved the ID generation method to the actual `SynonymRule` class

**Review by kderusso — APPROVED** [review]

**Comment by carlosdelest** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml (hunk 1 above)
Comment (mayya-sharipova) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml:22]: For this check we can use `is_true:`, for example: 

` - is_true: synonyms_set.0.id`

which means `synonyms_set.0.id` exists and is not `null` or `empty`

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml [hunk 3]
```diff
@@ -3,6 +3,25 @@ setup:
       version: " - 8.8.99"
       reason: Introduced in 8.9.0
 
+---
+"Create synonyms with no ID creates ID automatically":
```
Comment (mayya-sharipova) [inline rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml:7]: Should this test be from 8.10 version?

Comment (carlosdelest) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml:7]: Yes, I'll include it 👍

File: rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml (hunk 1 above)
Comment (carlosdelest) [reply rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/10_synonyms_put.yml:25]: TIL, thanks!

**Review by mayya-sharipova — APPROVED** [review]
@carlosdelest Thanks for iterating Carlos, this PR LGTM!

**Pull request merged**